unattended-upgrade

```diff
@@ -422,6 +428,14 @@ def signal_handler(signal, frame):
     SIGNAL_STOP_REQUEST = True
 
 
+def log_once(msg):
```

Comment:
This is fine - I personally would probably do it slightly different, I think its slightly nicer to have a `on_ac_power()` function and then on startup this can be dynamically patched. Something like:
```
import subprocess
import shutil

if not shutil.which("on_ac_power"):
    print("Checking if system ...")
    global on_ac_power
    on_ac_power = lambda: True


def on_ac_power():
    return subprocess.call("on_ac_power") == 1


if __name__ == "__main__":
    print(on_ac_power())
```

Reply:
I did consider something like that but please note that with the proposed code the warning is not shown if the check is not configured. Factoring out checks from should stop would indeed make the code easier to read, but I'd to that later.